### ISSUE TITLE
Create a "How to get involved in Jakarta EE" page 

### ISSUE THREAD

**Issue by shabnammayel** [jakartaee/jakarta.ee#1423]
Please create a "How to get involved in Jakarta EE" page that outlines various methods for developers to get involved in and contribute to Jakarta EE. The mockup is here:

https://projects.invisionapp.com/share/AX10M9QL74RN#/screens/462733154

Please find the content for the Testimonials section here (we need to list them on rotation when the page is refreshed): 

https://docs.google.com/document/d/19A55nEm8mwQcZrGOSZqbNUModB9e5MVrybl_k0zWBsI/edit

We can use these quotes that I extracted from previous "Featured Committer Profile" in the EF community newsletters dedicated to Jakarta EE so far. But I will also reach out to the community mailing lists to invite the contributors and committers of the WG to dedicate a quote if interested for us to add to this page later.

Please note that for the contributors section at the bottom, the idea is to feature and pull the images of all contributors and committers of the EE4J on that section and they should change when the page refreshes.

**Comment by ericpoirier** [comment]
Is there a deadline to implement this new page?

Testimonials:
Are those new testimonials to be added to our Testimonials data file? Or are those testimonials only for this new page?

Contributors section:
Can you confirm this is the list of users you'd like to se showing up in the Contributors section
https://projects.eclipse.org/projects/ee4j/who

**Comment by shabnammayel** [comment]
yes,  the deadline is to implement by the end of Q1. 

Testimonials: Those new testimonials are only for this page. 

That page only lists the PMCs of the project. the data needs to be pulled from the database. I think @waynebeaton can help 

FYI the two videos are currently uploaded on YouTube and are unlisted. we will make them public before going live

**Comment by ericpoirier** [comment]
> yes, the deadline is to implement by the end of Q1.

Sounds good

> Testimonials: Those new testimonials are only for this page.

I'll create  categories for the testimonials so that I can get specific testimonials for this page only.

> That page only lists the PMCs of the project. the data needs to be pulled from the database. I think @waynebeaton can help

Ok I think we can get that information from the API on the PMI.

> FYI the two videos are currently uploaded on YouTube and are unlisted. we will make them public before going live

I'll get a placeholder video for now and we can change it when the time comes.


Also I have another question. Where in the menu should we put this new page?

Thanks.

**Comment by TanjaObradovic** [comment]
@shabnammayel and @ericpoirier 
for any query that would require input from @waynebeaton and his team, please log an issue at https://gitlab.eclipse.org/eclipsefdn/infrazilla/-/issues

**Comment by ericpoirier** [comment]
@shabnammayel Can you also share the text content for this page?

Thanks.

**Comment by ericpoirier** [comment]
@shabnammayel @meKokabi I will also need all the graphic assets (logos, icons, pictures)

**Comment by waynebeaton** [comment]
> That page only lists the PMCs of the project. the data needs to be pulled from the database. I think @waynebeaton can help

What, specifically, do you want to show? Current committers? Everybody who has ever contributed to a project? What projects (just specification projects, or all EE4J projects)?

**Comment by shabnammayel** [comment]
We want to show current committers and contributors under EE4J projects. @waynebeaton

**Comment by shabnammayel** [comment]
here is the content doc to use @ericpoirier : https://docs.google.com/document/d/1Clh9P5k9z0aVCc7Z7ztwFXmq-DL2UXxlVm20Av-Y4Io/edit
@excentrickristy can you please share the graphic assets on the issue? thanks

**Comment by ericpoirier** [comment]
> here is the content doc to use @ericpoirier : https://docs.google.com/document/d/1Clh9P5k9z0aVCc7Z7ztwFXmq-DL2UXxlVm20Av-Y4Io/edit @excentrickristy can you please share the graphic assets on the issue? thanks

Thanks.

In the mockup, there seems to be videos. Can you post the links of those videos? I don't see them in the google doc.

**Comment by excentrickristy** [comment]
Here are the assets: https://www.dropbox.com/s/25x42wl3ammkc8s/Jakarta-Get%20Involved.zip?dl=0
Video links:
https://drive.google.com/file/d/12uK_qW7--ANuSAxqEFfeNkvN7AhMPXp-/view?usp=sharing
https://drive.google.com/file/d/1ys6LDjttEFoJ69VywEY8jL3KzwoRHS4i/view

**Comment by ericpoirier** [comment]
> Here are the assets: https://www.dropbox.com/s/25x42wl3ammkc8s/Jakarta-Get%20Involved.zip?dl=0 Video links: https://drive.google.com/file/d/12uK_qW7--ANuSAxqEFfeNkvN7AhMPXp-/view?usp=sharing https://drive.google.com/file/d/1ys6LDjttEFoJ69VywEY8jL3KzwoRHS4i/view

@shabnammayel We should probably put those videos on YouTube.

**Comment by ericpoirier** [comment]
> Here are the assets: https://www.dropbox.com/s/25x42wl3ammkc8s/Jakarta-Get%20Involved.zip?dl=0 Video links: https://drive.google.com/file/d/12uK_qW7--ANuSAxqEFfeNkvN7AhMPXp-/view?usp=sharing https://drive.google.com/file/d/1ys6LDjttEFoJ69VywEY8jL3KzwoRHS4i/view

@shabnammayel @meKokabi The provided assets are in an Illustrator file which I can't open on my new computer. Are you able to extract the logos,. icons, images so that I can use them for the web page?

Thanks.

**Comment by waynebeaton** [comment]
Here's a query that you can run against the `dashboard` database to get the committer id (when available), name, email addresses, and committer status of all of the committers and contributors who have authored at least one commit this year. I assume that @chrisguindon can pick and choose the columns that you actually want and sort out how to change the date range in the query.

````
select distinct 
    cm.id,
    if(cm.id is not null, concat(cm.first, ' ', cm.last), gca.name) as name,
    if(cm.email is not null, cm.email,gca.email) as email, 
    if(cp.id is null, "no","yes") as isCommitter 
from Project as p join GitRepo as gr on p.id=gr.project 
    join GitCommit as gc on gr.path=gc.path 
    join GitCommitAuthor as gca on gc.ref=gca.ref 
    left join CommitterEmail as ce on gca.email=ce.email 
    left join Committer as cm on ce.id=cm.id 
    left join CommitterProject as cp on cm.id=cp.id 
        and p.id=cp.project 
        and gc.date between cp.activeDate and cp.inactiveDate 
where p.parentId='ee4j' 
    and gc.date between date('2022-01-01') and now() 
    and gca.email not like '%-bot@eclipse.org'
    and gca.email not like 'dependabot%' 
    and gca.email != 'noreply@github.com'
````
Note that I am using a strict definition of committer and that somebody is listed as a committer only when they hold committer status on the project that they are actually committing to (so if somebody is a committer on project A, but has contributed to project B, they'd show up as a contributor).

Note that you'll notice some cases where two rows are obviously the same person, but they've used some other email address in their commit. I opportunistically fix these when I notice them (the database scheme has a means of supporting a committer having multiple email addresses). If the person is a committer, we can probably collapse them in data. If they are not a committer it's a little harder.

**Comment by ericpoirier** [comment]
@waynebeaton can't I simply use the following API call to retrieve committers and contributors of the EE4J project:

https://projects.eclipse.org/api/projects/ee4j

**Comment by ivargrimstad** [comment]
> @waynebeaton can't I simply use the following API call to retrieve committers and contributors of the EE4J project:
> 
> https://projects.eclipse.org/api/projects/ee4j

That would only give you the committers on the top-level project. You would have to drill down to all the sub-projects of EE4J in order to get all committers...

**Comment by ericpoirier** [comment]
> > @waynebeaton can't I simply use the following API call to retrieve committers and contributors of the EE4J project:
> > https://projects.eclipse.org/api/projects/ee4j
> 
> That would only give you the committers on the top-level project. You would have to drill down to all the sub-projects of EE4J in order to get all committers...

Ok then we will need to look into creating a new API endpoint on the PMI that will give us what we need.

**Comment by ericpoirier** [comment]
@shabnammayel, currently looking at the Contributors section of the following mockup:

https://projects.invisionapp.com/share/AX10M9QL74RN#/screens/462733154

Which logo are we adding for each contributors? For example, if the person contributed to the Glassfish project, are we showing the Glassfish logo. Or are we always printing the Jakarta EE logo?

Also, if the person contributed to multiple projects, which logo are we choosing?

Also, not all projects have their own logos. Many have the Eclipse orb or the Incubating logo which doesn't tell us much about the project if we're only printing the logo.

Also are we linking the contributor's name to something?
[Example: https://projects.eclipse.org/user/8408](https://projects.eclipse.org/user/5806)
Or: [https://accounts.eclipse.org/users/epoirier](https://accounts.eclipse.org/users/epoirier)

**Comment by waynebeaton** [comment]
> > @waynebeaton can't I simply use the following API call to retrieve committers and contributors of the EE4J project:
> > https://projects.eclipse.org/api/projects/ee4j
> 
> That would only give you the committers on the top-level project. You would have to drill down to all the sub-projects of EE4J in order to get all committers...

The query that I've provided shows the people who actually contributed something. That is, it will produce a list of people who have contributed at least one commit to at least one repository belonging to at least one EE4J subproject in the specified period. Committers who are not active in the specified period will be excluded. Further, this will include contributors, whether they are designated in the PMI as a contributor or not.

If this is not what we want, let me know.

**Comment by ericpoirier** [comment]
> > > @waynebeaton can't I simply use the following API call to retrieve committers and contributors of the EE4J project:
> > > https://projects.eclipse.org/api/projects/ee4j
> > 
> > 
> > That would only give you the committers on the top-level project. You would have to drill down to all the sub-projects of EE4J in order to get all committers...
> 
> The query that I've provided shows the people who actually contributed something. That is, it will produce a list of people who have contributed at least one commit to at least one repository belonging to at least one EE4J subproject in the specified period. Committers who are not active in the specified period will be excluded. Further, this will include contributors, whether they are designated in the PMI as a contributor or not.
> 
> If this is not what we want, let me know.

That's a good question.

If we want to link those contributors somewhere (on accounts or pmi) I think we should validate that the user have an eclipse account before showing them on Jakarta EE?

@shabnammayel

**Comment by shabnammayel** [comment]
The videos were on YouTube but unlisted. They are public now: https://www.youtube.com/watch?v=TiegAZbqSpI&t=11s
https://www.youtube.com/watch?v=eSY6Uo62IH8
We can remove the logo from the boxes in the contributors section and just include their pics, name and last name

**Comment by ericpoirier** [comment]
> The videos were on YouTube but unlisted. They are public now: https://www.youtube.com/watch?v=TiegAZbqSpI&t=11s https://www.youtube.com/watch?v=eSY6Uo62IH8 We can remove the logo from the boxes in the contributors section and just include their pics, name and last name

Awesome, thanks.

Also is it possible for @meKokabi to extract the images that we need from the Illustrator file that @excentrickristy provided?

Thanks.

**Comment by chrisguindon** [comment]
@shabnammayel Why is there a deadline to complete this by end of the quarter? 

This issue is not listed in our list of objectives for Q1: https://github.com/jakartaee/jakarta.ee/issues/1236


@ericpoirier Regarding the contributor API. the community created a script for that for building the releases pages. Can you take a look at that before implementing anything new in the PMI? https://github.com/jakartaee/jakarta.ee#contributors-list-optional

**Comment by ericpoirier** [comment]
> @shabnammayel Why is there a deadline to complete this by end of the quarter?
> 
> This issue is not listed in our list of objectives for Q1: #1236
> 
> @ericpoirier Regarding the contributor API. the community created a script for that for building the releases pages. Can you take a look at that before implementing anything new in the PMI? https://github.com/jakartaee/jakarta.ee#contributors-list-optional

@chrisguindon The script generates the following files:
https://github.com/jakartaee/jakarta.ee/blob/src/data/contributors/jakarta-ee-9.1.json
https://github.com/jakartaee/jakarta.ee/blob/src/data/contributors/jakarta-ee-9.json

@ivargrimstad @shabnammayel are those list good for what we need? If so, can we use the latest release file (jakarta-ee-9.1.json) or would we have to merge them together?

Thanks.

**Comment by shabnammayel** [comment]
Yes @waynebeaton sounds good!

The script that community created for the release pages only included folks involved in that release version and I believe they were only committers. but in this case what Wayne explained above is what we are looking for @chrisguindon 
@ericpoirier All committers and contributors should already have an EF account before contributing any work.

**Comment by chrisguindon** [comment]
@shabnammayel You did not answer my first question. 

Why is there a deadline to complete this by end of the quarter?  To be honest, we definitely don't have the time to get this done by the end of the month if we need to create a new API endpoint.

If we are going to create new deadline, I would really appreciate that you discuss them with me first.

**Comment by shabnammayel** [comment]
We were aiming for end of Q2! it's not set in stone! with the new requirement to create a new API, we can discuss and set a deadline to get this page implemented @chrisguindon

**Comment by chrisguindon** [comment]
Thanks @shabnammayel! Q2 is definitely doable for us!

**Comment by shabnammayel** [comment]
I think we can work towards implementing the page excluding the contributors section and add that later as well. thanks

**Comment by meKokabi** [comment]
@ericpoirier As requested, please find attached, the visual assets:
[JakartaGetInvolved-Assets.zip](https://github.com/jakartaee/jakarta.ee/files/8217795/JakartaGetInvolved-Assets.zip)

**Comment by shabnammayel** [comment]
@ericpoirier are there any update on this page's implementation? As I mentioned before we can go live with the page without the "Contributors" section first and implement that section in Q2. thanks

**Comment by ericpoirier** [comment]
> @ericpoirier are there any update on this page's implementation? As I mentioned before we can go live with the page without the "Contributors" section first and implement that section in Q2. thanks

It's in my plans to work on that page this week.

**Comment by ericpoirier** [comment]
> @ericpoirier As requested, please find attached, the visual assets: [JakartaGetInvolved-Assets.zip](https://github.com/jakartaee/jakarta.ee/files/8217795/JakartaGetInvolved-Assets.zip)

Hi @meKokabi can you provide the white icons without the blue background. I will also need the orange icons from the mockup.

Also missing the Testimonials background.

Thanks.

**Comment by ericpoirier** [comment]
> The videos were on YouTube but unlisted. They are public now: https://www.youtube.com/watch?v=TiegAZbqSpI&t=11s https://www.youtube.com/watch?v=eSY6Uo62IH8 We can remove the logo from the boxes in the contributors section and just include their pics, name and last name

Can you provide the "How to contribute" link as well?

Thanks.

**Comment by shabnammayel** [comment]
We had to unlist the videos as they have links to the how to get involved page that is not live yet. I add the links to both above but I think they were pasted very closely so a bit hard to notice :) 

https://www.youtube.com/watch?v=TiegAZbqSpI&t=11s 
https://www.youtube.com/watch?v=eSY6Uo62IH8

**Comment by meKokabi** [comment]
@ericpoirier Here You Are! Let me know if you. need any more assets
[Missing Assets-png-svg.zip](https://github.com/jakartaee/jakarta.ee/files/8381958/Missing.Assets-png-svg.zip)
.

**Comment by ericpoirier** [comment]
> @ericpoirier Here You Are! Let me know if you. need any more assets [Missing Assets-png-svg.zip](https://github.com/jakartaee/jakarta.ee/files/8381958/Missing.Assets-png-svg.zip) .

Thanks.

Can you also provide the orange icons from the mockup?

**Comment by meKokabi** [comment]
Already done! it is in the SVG folder.

**Comment by ericpoirier** [comment]
> Already done! it is in the SVG folder.

Looking into the svg folder and I am missing the following orange icons:

- Social Media
- Slack
- Paper with lines on it

Thanks.

**Comment by meKokabi** [comment]
> > Already done! it is in the SVG folder.
> 
> Looking into the svg folder and I am missing the following orange icons:
> 
> * Social Media
> * Slack
> * Paper with lines on it
> 
> Thanks.


Please find attached and let me know if anything still missed
[OrangeIcons.zip](https://github.com/jakartaee/jakarta.ee/files/8382219/OrangeIcons.zip)

**Comment by ericpoirier** [comment]
@shabnammayel please review the following page:

https://deploy-preview-1428--jakartaee.netlify.app/community/get-involved/

Let me know if you'd like to change anything.

Thanks.

**Comment by shabnammayel** [comment]
for the first box, can we jump to the social media section of the page in the footer when they click on [Join us on Social Media](https://jakarta.ee/connect/)? currently it goes to the top of the page and it's hard to find the handles.

**Comment by shabnammayel** [comment]
Pictures of some of the folks in the testimonial section is missing, can you please get their featured committer profile image that are added in this doc? https://docs.google.com/document/d/19A55nEm8mwQcZrGOSZqbNUModB9e5MVrybl_k0zWBsI/edit

When you open the doc you can see a link attached on their names that takes you to their committer profiles in the EF community newsletter. we can use those images. thanks

**Comment by shabnammayel** [comment]
<img width="1006" alt="Screen Shot 2022-03-30 at 2 43 16 PM" src="https://user-images.githubusercontent.com/51449073/160908161-f8371a10-3e22-40f8-a8c1-2a502211c63e.png">
 
Can we please add the page on the homepage as well. in the Get involved box as "How to Get Involved" above the Join the Working Group button

**Comment by shabnammayel** [comment]
Also can we change the link in the footer to go to this page instead of the EF page?
<img width="683" alt="Screen Shot 2022-03-30 at 2 45 57 PM" src="https://user-images.githubusercontent.com/51449073/160908679-4ed77a49-00eb-489e-9cc0-d181158ad6fd.png">

**Comment by ericpoirier** [comment]
@shabnammayel I updated staging as per your feedback.

https://deploy-preview-1428--jakartaee.netlify.app/community/get-involved/

Let me know if there's anything else.

**Comment by shabnammayel** [comment]
Looks good! Thanks. Can we sort the items in the homepage boxes on an alphabetic order? 

Also please let me know when we are ready to go live to make the videos public. thanks @ericpoirier

**Comment by ericpoirier** [comment]
> Looks good! Thanks. Can we sort the items in the homepage boxes on an alphabetic order?

I fixed the order on the home page:

https://deploy-preview-1428--jakartaee.netlify.app/

> Also please let me know when we are ready to go live to make the videos public. thanks @ericpoirier

Ok will do.

**Comment by shabnammayel** [comment]
LGTM! thanks
